### ISSUE TITLE
Reducing dependencies size

### ISSUE THREAD

**Issue by kylepierce** [awslabs/aws-data-wrangler#528]
**Is your idea related to a problem? Please describe.**
We have been using this package with aws lambda. Worked fine until recently had to add a package. Now, ufortunately its hitting the max unzip file size. Top 5 packages in my lambda come from this package. The smallest (numpy 24mb) is double the size of the next largest package (lxml 12mb).

PyArrow - 56mb
Botocore - 53mb
Pandas - 54mb
Numpy libs - 33mb
Numpy - 24mb

Basically 200mb just to use this package. 

**Describe the solution you'd like**
Clearly this is a very valuable package. Maybe there can be a way to compress those packages to be around 150mb?

**Comment by igorborgest** [comment]
Hi @kylepierce our ["pre-built" lambda layer](https://aws-data-wrangler.readthedocs.io/en/stable/install.html#aws-lambda-layer) already has [several optimizations](https://github.com/awslabs/aws-data-wrangler/blob/main/building/lambda/build-lambda-layer.sh). Did you try to use it keeping your other dependencies in a second layer?

**Comment by kylepierce** [comment]
We are currently zipping the entire folder (using sam) and uploading to s3. If we upload without those dependencies will we be able to access them without changing the code?

**Comment by igorborgest** [comment]
Yes, it should work.
